secboot/secboot_tpm.go

```diff
@@ -780,7 +783,101 @@ func resealKeysWithTPMImpl(params *resealKeysWithTPMParams, newPCRPolicyVersion
 
 var resealKeysWithTPM = resealKeysWithTPMImpl
 
-func buildPCRProtectionProfile(modelParams []*SealKeyModelParams, allowInsufficientDmaProtection bool) (*sb_tpm2.PCRProtectionProfile, error) {
+// buildPCRProtectionProfile constructs a PCR protection profile used for sealing
+// encryption keys.
+//
+// If checkResult is non-nil, it provides the PCR configuration recommended by
+// the pre-install check. In this mode, allowInsufficientDmaProtection is ignored,
+// because all relevant information is carried by checkResult.
+//
+// If checkResult is nil, the function falls back to the legacy profile-generation
+// path, which relies on static assumptions about the system’s boot configuration.
+// In this legacy mode, allowInsufficientDmaProtection determines whether systems
+// lacking sufficient DMA protection are still permitted.
+func buildPCRProtectionProfile(modelParams []*SealKeyModelParams, checkResult *PreinstallCheckResult, allowInsufficientDmaProtection bool) (*sb_tpm2.PCRProtectionProfile, error) {
+	if checkResult == nil {
+		return buildPCRProtectionProfileLegacy(modelParams, allowInsufficientDmaProtection)
+	}
+
+	// build EFI load image event trees
+	loadSequences := sb_efi.NewImageLoadSequences()
+	for _, mp := range modelParams {
+		efiLoadActivities, err := buildEFILoadActivitiesFromChains(
+			mp.EFILoadChains,
+			sb_efi.SnapModelParams(mp.Model),
+			sb_efi.KernelCommandlineParams(mp.KernelCmdlines...),
+		)
```

Comment:
Minor comment, but I wonder whether `buildEFILoadActivitiesFromChains` should just accept the `SealKeyModelParams` instead?